File: examples/project-template/src/main/java/be/kuleuven/csa/ProjectMain.java
package be.kuleuven.csa;

import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;

public class ProjectMain extends Application {

    private static Stage rootStage;

    public static Stage getRootStage() {
        return rootStage;
    }

    @Override
    public void start(Stage stage) throws Exception {
        rootStage = stage;
        Parent root = FXMLLoader.load(getClass().getClassLoader().getResource("csamain.fxml"));

        Scene scene = new Scene(root);

        stage.setTitle("CSA Administratie hoofdscherm");
        stage.setScene(scene);
        stage.show();
    }

    public static void main(String[] args) {
        launch();
    }
}

File: examples/kotlin/project-template/src/main/kotlin/be/kuleuven/csa/ProjectMain.kt
package be.kuleuven.csa

import javafx.application.Application
import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.stage.Stage

var rootStage: Stage? = null

class ProjectMain : Application() {
    override fun start(primaryStage: Stage?) {
        rootStage = primaryStage
        val scene = Scene(loadFXML("/csamain.fxml"))
        with(primaryStage!!) {
            this.scene = scene
            title = "CSA Administratie hoofdscherm"
            show()
        }
    }

    private fun loadFXML(fxml: String): Parent {
        val resource = this::class.java.getResource(fxml)
        return FXMLLoader(resource).load()
    }
}

fun main(args: Array<String>) {
    Application.launch(ProjectMain::class.java, *args)
}